platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -184,17 +184,14 @@ class ConfigurationCacheProblems(
     }
 
     override fun forTask(task: Task): ProblemsListener {
-        val degradationReasons = degradationController.getDegradationReasonsForTask(task)
+        val degradationReasons = degradationController.degradationDecision.degradationReasonForTask(task)
```

Comment:
💅 Dedupe access to `degradationDecision` in this class

Reply:
Fixed